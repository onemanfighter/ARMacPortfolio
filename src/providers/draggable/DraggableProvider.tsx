import Draggable from 'react-draggable';
import { useWindowDimensions } from '@hooks';
import { DraggableProps } from './types';

const DraggableProvider = ({
  children,
  position,
  maximized,
  onPositionChange,
}: DraggableProps) => {
  const { width, height } = useWindowDimensions();

  console.log(position);

  const positionHandler = maximized
    ? {}
    : {
        onStop: (e: any, data: any) => {
          const { x, y } = data;
          onPositionChange({ x, y });
        },
      };

  return (
    <Draggable
      handle=".handle"
      bounds={{ left: 0, top: 0, right: width, bottom: height }}
      positionOffset={{ x: 0, y: 28 }}
      defaultPosition={maximized ? position : { x: 0, y: 0 }}
      position={!maximized ? position : { x: 0, y: 0 }}
      grid={[3, 3]}
      scale={1}
      {...positionHandler}
    >
      {children}
    </Draggable>
  );
};

export default DraggableProvider;
