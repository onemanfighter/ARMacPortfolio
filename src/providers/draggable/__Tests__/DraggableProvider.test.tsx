import { render } from '@testing-library/react';
import DraggableProvider from '../DraggableProvider';
import { Box } from '@chakra-ui/react';

describe('Draggable provider', () => {
  it('should render correctly', () => {
    const { container } = render(
      <DraggableProvider position={{ x: 0, y: 0 }}>
        <Box>test</Box>
      </DraggableProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
