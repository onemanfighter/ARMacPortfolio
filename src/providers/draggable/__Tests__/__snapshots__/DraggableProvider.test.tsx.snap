// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draggable provider should render correctly 1`] = `
<div>
  <div
    class="react-draggable css-0"
    style="transform: translate(0px,0px);"
  >
    test
  </div>
</div>
`;
